app/src/main/java/com/gophillygo/app/EventsListActivity.java

```diff
@@ -74,16 +78,27 @@ protected void onCreate(Bundle savedInstanceState) {
         LiveData<Resource<List<EventInfo>>> data = viewModel.getEvents();
         data.observe(this, destinationResource -> {
             if (destinationResource != null && destinationResource.status.equals(Status.SUCCESS) &&
-                    destinationResource.data != null && !destinationResource.data.isEmpty()) {
-
-                eventsListView = findViewById(R.id.events_list_recycler_view);
-                EventsListAdapter adapter = new EventsListAdapter(this, destinationResource.data, this);
-                eventsListView.setAdapter(adapter);
-                eventsListView.setLayoutManager(layoutManager);
+                    destinationResource.data != null) {
```

Comment:
Are the `isEmpty` checks no longer needed? There are some places that expect at least one attraction, as the first one is used to check some properties before setting distances.

Reply:
This is vestigial from when we were filtering using SQL and the data list could be empty (and your comment explains why distances were being cleared when I filtered to an empty list and then reset).

I'll add the `isEmpty()` checks back in.